runtime/compiler/optimizer/UnsafeFastPath.cpp

```diff
@@ -696,6 +797,8 @@ int32_t TR_UnsafeFastPath::perform()
          // Handle VarHandle operation methods
          bool isStatic = false;
          TR::RecognizedMethod callerMethod = methodSymbol->getRecognizedMethod();
+         if (recognizedVarHandleOpMethod != TR::unknownMethod)
+            callerMethod = recognizedVarHandleOpMethod;
```

Comment:
I suppose we look up the stack for the benefit of the various `ByteBufferHandle` classes? It looks like those ones call through `ScopedMemoryAccess`, whereas all the others I saw called `Unsafe` methods directly

Reply:
Yes, this is to handle cases where the VH operation methods don't call Unsafe methods directly.

Reply:
Other than some operations calling Unsafe methods through ScopedMemoryAccess, there is also calls to other helper methods often defined within the same class. For example, `getAndAddRelease` in `VarHandleByteArrayAsInts$ArrayHandle`, which calls the Unsafe method directly when no endian conversion is needed, but calls the Unsafe method through `getAndAddConvEndianWithCAS` otherwise.